app/models/v2/report.rb

```diff
@@ -44,12 +86,25 @@ def percent_compliant
     has_many :report_systems, class_name: 'V2::ReportSystem', dependent: nil # rubocop:disable Rails/InverseOf
     has_many :systems, class_name: 'V2::System', through: :report_systems
     has_many :reported_systems, class_name: 'V2::System', through: :test_results, source: :system, dependent: nil
+    # rubocop:disable Rails/InverseOf
+    has_many :reporting_and_non_reporting_systems, lambda {
+      joins(
+        arel_table.join(V2::System.arel_table, AN::InnerJoin)
+                  .on(V2::System.arel_table[:id].eq(V2::ReportSystem.arel_table[:system_id]))
```

Comment:
If you pick the `class_name` correctly (see below) then you don't need to specify the join with the systems, it will be assumed automatically.